static/src/javascripts/projects/commercial/modules/hosted/gallery.spec.js

```diff
@@ -0,0 +1,155 @@
+        gallery.handleKeyEvents({ keyCode: 73 });
+
+        expect(classListFor('js-gallery-caption-bar')).toEqual(
+            expect.stringContaining('hosted-gallery--show-caption')
+        );
+
+        gallery.handleKeyEvents({ keyCode: 73 });
+
+        expect(classListFor('js-gallery-caption-bar')).not.toEqual(
+            expect.stringContaining('hosted-gallery--show-caption')
+        );
+
+        done();
+    });
+
+    it('should open and close the onward journey on click', done => {
+        const button: any = document.querySelector(
+            '.js-hosted-gallery-oj-close'
+        );
+        button.click();
+
+        expect(classListFor('js-hosted-gallery-oj')).toEqual(
+            expect.stringContaining('minimise-oj')
+        );
+
+        button.click();
+
+        expect(classListFor('js-hosted-gallery-oj')).not.toEqual(
+            expect.stringContaining('minimise-oj')
+        );
+
+        done();
+    });
+
+    it('should log navigation in GA when using arrow key navigation', done => {
+        gallery.handleKeyEvents({ keyCode: 40, preventDefault: noop });
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('KeyPress:down - image 2');
+        gallery.handleKeyEvents({ keyCode: 39, preventDefault: noop });
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('KeyPress:right - image 3');
+        gallery.handleKeyEvents({ keyCode: 38, preventDefault: noop });
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('KeyPress:up - image 2');
+        gallery.handleKeyEvents({ keyCode: 37, preventDefault: noop });
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('KeyPress:left - image 1');
+        done();
+    });
+
+    it('should log navigation in GA when clicking through images', done => {
+        gallery.initScroll.call(gallery);
+        (document.querySelector('.inline-arrow-down'): any).click();
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('Click - image 2');
+        (document.querySelector('.inline-arrow-up'): any).click();
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('Click - image 1');
+        done();
+    });
+
+    it('should log navigation in GA when scrolling through images', done => {
+        const nativeHTMLElement = window.HTMLElement;
+        window.HTMLElement = function() {
+            this.scrollTop = 20;
+            this.scrollHeight = 30;
+        };
+        gallery.fadeContent({
+            target: new window.HTMLElement(),
+        });
+        expect(
+            interactionTracking.trackNonClickInteraction
+        ).toHaveBeenCalledWith('Scroll - image 3');
+        done();
```

Comment:
The tests in this spec don't need to accept or call the `done` callback as they are not async